charmhelpers/fetch/ubuntu.py

```diff
@@ -189,17 +190,185 @@
 CMD_RETRY_COUNT = 3  # Retry a failing fatal command X times.
 
 
+def dpkg_list(packages):
+    """Get data from system dpkg database for package.
+
+    Note that one of the main purposes for this function is to avoid dependency
+    on the ``python-apt`` package.
+
+    The indicated package is a wrapper around the ``apt`` C++ library which is
+    tightly connected to the version of the distribution it is shipped on.
+    This in turn makes it incredibly hard to distribute as a wheel for a charm
+    that supports a large span of distro versions.
+
+    :param packages: Packages to get data from
+    :type packages: list[str]
```

Comment:
This is picky, but it's `List[str]`

(and I believe it's because it's a class in the types module).

Reply:
:+1: